File: trunk/ShadowFrameworkJS/src/shadow/pipeline/SFIndexRange.js

function SFIndexRange(position,size) {
	this.position = position;
	this.size = size;
}


SFIndexRange.prototype["getPosition"]=function(){
	return this.position;
};

SFIndexRange.prototype["getSize"]=function(){
	return this.size;
};

SFIndexRange.prototype["setPosition"]=function(position){
	this.position=position;
};

SFIndexRange.prototype["setSize"]=function(size){
	this.size=size;
};

SFIndexRange.prototype["insertIndex"]=function(index){
	if(this.position==-1){
		this.position=index;
		this.size=1;
	}else{
		var last=this.position+this.size;
		if(index<this.position){
			this.position=index;
			this.size=last-this.position;
		}else if(index>=this.position+this.size){
			this.size=index-this.position+1;
		}
	}
};

File: SFDataApplicationTestC++/shadow/pipeline/SFIndexRange.cpp
#include "SFIndexRange.h"

namespace sf{

SFIndexRange::SFIndexRange(int position, int size) {
    this->position = position;
    this->size = size;
}

int SFIndexRange::getPosition() {
    return position;
}
void SFIndexRange::setPosition(int position) {
    this->position = position;
}
int SFIndexRange::getSize() {
    return size;
}
void SFIndexRange::setSize(int size) {
    this->size = size;
}

void SFIndexRange::insertIndex(int index){
    if(this->position==-1){
        this->position=index;
        this->size=1;
    }else{
        int last=position+size;
        if(index<position){
            position=index;
            size=last-position;
        }else if(index>=last){
            size=index-position+1;
        }
    }
}

}
